tasks/eth/018-granite-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,179 @@
+    address constant expectedFDG = 0xA6f3DFdbf4855a43c529bc42EDE96797252879af;
+    address constant expectedPermissionedFDG = 0x050ed6F6273c7D836a111E42153BC00D0380b87d;
+
+    Types.ContractSet proxies;
+
+    /// @notice Sets up the contract
+    function setUp() public {
+        proxies = _getContractSet();
+    }
+
+    function getCodeExceptions() internal view override returns (address[] memory) {
+        // Safe owners will appear in storage in the LivenessGuard when added, and they are allowed
+        // to have code AND to have no code.
+        address[] memory securityCouncilSafeOwners = securityCouncilSafe.getOwners();
+
+        // To make sure we probably handle all signers whether or not they have code, first we count
+        // the number of signers that have no code.
+        uint256 numberOfSafeSignersWithNoCode;
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            if (securityCouncilSafeOwners[i].code.length == 0) {
+                numberOfSafeSignersWithNoCode++;
+            }
+        }
+
+        // Then we extract those EOA addresses into a dedicated array.
+        uint256 trackedSignersWithNoCode;
+        address[] memory safeSignersWithNoCode = new address[](numberOfSafeSignersWithNoCode);
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            if (securityCouncilSafeOwners[i].code.length == 0) {
+                safeSignersWithNoCode[trackedSignersWithNoCode] = securityCouncilSafeOwners[i];
+                trackedSignersWithNoCode++;
+            }
+        }
+
+        // Here we add the standard (non Safe signer) exceptions.
+        address[] memory shouldHaveCodeExceptions = new address[](4 + numberOfSafeSignersWithNoCode);
+
+        shouldHaveCodeExceptions[0] = systemConfigOwner;
+        shouldHaveCodeExceptions[1] = batchSenderAddress;
+        shouldHaveCodeExceptions[2] = p2pSequencerAddress;
+        shouldHaveCodeExceptions[3] = batchInboxAddress;
+
+        // And finally, we append the Safe signer exceptions.
+        for (uint256 i = 0; i < safeSignersWithNoCode.length; i++) {
+            shouldHaveCodeExceptions[4 + i] = safeSignersWithNoCode[i];
+        }
+
+        return shouldHaveCodeExceptions;
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](6);
+        allowed[0] = address(proxies.DisputeGameFactory);
+        allowed[1] = address(ownerSafe);
+        allowed[2] = address(securityCouncilSafe);
+        allowed[3] = address(fndSafe);
+        allowed[4] = expectedLivenessGuard;
+        allowed[5] = proxies.AnchorStateRegistry;
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory /* simPayload */ )
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
```

Comment:
This will prevent the issue in #268. I ran a smoke test locally so this should be good to commit.

```suggestion
        if (msg.sig == this.approveJson.selector) {
            console.log("Skipping assertions on the approval call");
            return;
        }

        console.log("Running post-deploy assertions");
```

Reply:
added. Thanks.